### ISSUE TITLE
Check `Base.USE_GPL_LIBS` before loading GPL code

### ISSUE THREAD

**Issue by simsurace** [JuliaStats/Rmath.jl#86]
Could we check for the value of `Base.USE_GPL_LIBS` and only load the GPL code in this package (which seems to be all of it) when it evaluates to true? This would allow one to easily check whether a given application/package works when Julia is built without GPL libraries.

**Comment by peteristhegreat** [comment]
Agreed.  The license for this package should probably be changed to be the same as librmath (GPLv2) unless a special exception is requested/amended onto the license of librmath.

**Comment by ViralBShah** [comment]
Licensing clarified in b4a1bec779efe3311fac5d3d4ac9ff078983dae2

**Comment by ViralBShah** [comment]
@simsurace `Base.USE_GPL_LIBS` typically is only for the Julia distribution itself using GPL libs, since people want to build Julia without GPL libraries altogether sometimes. There are many GPL packages that you can use even if Julia itself was built without GPL dependencies, and those dependencies (mainly SuiteSparse) may be moved out of Julia in a future release.

I don't think it would be a good idea to make this package loading conditional on `Base.USE_GPL_LIBS`. If we want license wise restrictions (like never use GPL packages in this environment), that mechanism probably needs to be in Pkg.jl or something.
